src/runtime_src/xrt/device/hal2.h

```diff
@@ -652,6 +652,21 @@ class device : public xrt::hal::device
     return sysfs_path;
   }
 
+  virtual hal::operations_result<std::string>
+  getSubdevPath(const std::string& subdev, uint32_t idx)
+  {
+    if (!m_ops->mGetSubdevPath)
+      return hal::operations_result<std::string>();
+    constexpr size_t max_path = 256;
+    char path_buf[max_path];
+    if (m_ops->mGetSubdevPath(m_handle, subdev.c_str(), idx, path_buf, max_path)) {
+      return hal::operations_result<std::string>();
+    }
+    path_buf[max_path - 1] = '\0';
+    std::string sysfs_path = std::string(path_buf);
+    return sysfs_path;
+  }
+
```

Comment:
Ditto.

Reply:
change "sysfs_path" to more general one "path"